code/game/objects/items/dna_injector.dm

```diff
@@ -314,7 +314,7 @@
 		to_chat(user, "<span class='notice'>You can't modify [M]'s DNA while [M.p_theyre()] dead.</span>")
 		return FALSE
 
-	if(M.has_dna() && !(M.disabilities & NOCLONE))
+	if(M.has_dna() && !(RADIMMUNE in M.dna.species.species_traits) && !(M.disabilities & NOCLONE))

```

Comment:
Just a heads up, this isn't me being hacky. This is consistent with the other behavior:

https://github.com/OracleStation/OracleStation/blob/4b0a0450f6ec1d6923d05bcfc92a260cf4794c93/code/game/objects/items/dna_injector.dm#L35